stl/inc/chrono

```diff
@@ -617,27 +669,1289 @@ namespace chrono {
     };
 
     using high_resolution_clock = steady_clock;
+
+#if _HAS_CXX20
+    struct local_t {};
+
+    template <class _Duration>
+    using local_time    = time_point<local_t, _Duration>;
+    using local_seconds = local_time<seconds>;
+    using local_days    = local_time<days>;
+
+    struct last_spec {
+        explicit last_spec() = default;
+    };
+    inline constexpr last_spec last{};
+
+    class day {
+    public:
+        day() = default;
+        constexpr explicit day(unsigned int _Val) noexcept : _Day{static_cast<unsigned char>(_Val)} {}
+
+        constexpr day& operator++() noexcept {
+            ++_Day;
+            return *this;
+        }
+        constexpr day operator++(int) noexcept {
+            return day{_Day++};
+        }
+        constexpr day& operator--() noexcept {
+            --_Day;
+            return *this;
+        }
+        constexpr day operator--(int) noexcept {
+            return day{_Day--};
+        }
+
+        constexpr day& operator+=(const days& _Days) noexcept {
+            _Day += static_cast<unsigned char>(_Days.count());
+            return *this;
+        }
+        constexpr day& operator-=(const days& _Days) noexcept {
+            _Day -= static_cast<unsigned char>(_Days.count());
+            return *this;
+        }
+
+        _NODISCARD constexpr explicit operator unsigned int() const noexcept {
+            return _Day;
+        }
+        _NODISCARD constexpr bool ok() const noexcept {
+            return _Day >= 1 && _Day <= 31;

```

Comment:
Possible micro-optimization: what if we implemented this with `static_cast<unsigned char>(_Day - 1) <= 30`? Would that improve the optimized codegen, or are MSVC and LLVM already capable of performing that transformation?

I would be totally fine with filing an issue to investigate micro-optimizing this later, as it has no ABI or source compat impact.

Reply:
Our lord and saviour says they omit the same thing.

https://godbolt.org/z/sMz5h1